### DISCUSSION TITLE
Is there a way to cycle through screens?

### DISCUSSION THREAD

**Discussion by Azn9** [ppy/osu-framework#6330] [Q&A] [answered]
I created multiple Screen classes and one that extends from ScreenStack to handle the switches:
```C#
public partial class EditorSubScreens : ScreenStack
{
    private readonly ChartInfoScreen chartInfoScreen = new();
    private readonly TimingScreen timingScreen = new();

    public EditorSubScreens()
    {
        Push(chartInfoScreen); // Default screen
    }

    public void DisplayChartInfoScreen()
    {
        Exit();
        Push(chartInfoScreen);
    }

    public void DisplayTimingScreen()
    {
        Exit();
        Push(timingScreen);
    }
}
```

However, calling `DisplayTimingScreen` then `DisplayChartInfoScreen` (to go back to the old screen) throws `A screen should not be loaded before being pushed.`. I currently only have two screens, but at the end, I will have 4 or 5 of them and want the user to be able to switch between each without having to rebuild it entirely each time (which will be very costly for at least one of them).
How could I do that? Thanks in advance ;)

**Comment by peppy** [comment, marked as answer]
It sounds like you might want something that isn't a stack?

Something like this setup may give you a bit of inspiration: https://github.com/ppy/osu/blob/8a06914438fd3d1d8bdcc462c5fe196891bee2c7/osu.Game.Tournament/TournamentSceneManager.cs#L54

**Reply by Azn9** [reply]
Thanks for your reply; this is indeed what I was searching for! I used ScreenStacks because I didn't even think it could be done without them to be honest 😅 
Thank you for your time! <3
